### PR TITLE
Update network topology.  Mesh->small-world 

### PR DESCRIPTION
**Pull request by UnstoppableCash**
Also, update:
"It’s also unnecessary and wasteful of network resources to connect to more than a handful of nodes."

With:
Since mining nodes are incentivized to connect to other miners to maximize propagation speed and efficiency, a miner may have hundreds of connections.

### CODE REVIEW

**Comment by UnstoppableCash** [comment]
Substantiation for "miners with thousands of connections"

The [Coinscope paper](https://www.cs.umd.edu/projects/coinscope/coinscope.pdf), clear back in 2015 found that:

"_Another feature of the mainline Satoshi client is that
it permits a maximum of 125 active connections, but
we consistently see nodes that far exceed this maximum,
sometimes by a factor of nearly 80. This is not a singu-
lar phenomenon; we see these high degree nodes across
all runs of AddressProbe. That is, extremely high-degree
nodes are persistent over time._"

**Comment by gmaxwell** [comment]
Please don't propagate scammer Craig Wright's bullshit.  The existence of spy nodes is a well known unfortunate phenomena which in no way supports wright's bizarre claims which are completely out of wack with the obvious operation of the network (and even the clear claims of the work you cite!).

**Comment by mpapec** [comment]
Bitcoin fierce critic which found how wrong he was and went into bitcoin core dev team. Despite that still feeling unmistakable, and in position to give lessons from position of authority, and unshaken credibility. Personally would not believe what time is it from such sources.

**Comment by funbucks** [comment]
@gmaxwell go work on codecs; fake expert

**Comment by dymurray** [comment]
@gmaxwell Here are some links to understand the network topology of Bitcoin.

Bitcoin (BCH) is a “small world” network which is scale-free. 

https://www.researchgate.net/publication/262562539_Exploring_the_Bitcoin_Network
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3604768/
https://www.scribd.com/document/359522859/Bitcoin-Network-Topology-ELI5

**Comment by checksum0** [comment]
There's no scientific concensus over this, Even the scientist that worked on the paper with CSW is presenting all the results as "preliminary" and "early".

Let the peer review process continue before jumping to conclusions.

**Comment by dymurray** [comment]
@checksum0 CSW is not the only person to observe the small-world phenomena in Bitcoin. The paper I cite above was not authored by Craig Wright and uses historical data to show that the network exhibits a high clustering coefficient which indicates it is a small-world network.

It is also very intuitive as the economic incentives of Bitcoin lead miners to connect to as many other miners as possible. This makes the network a near-complete graph where almost the entire network is aware of transactions almost instantly.

Regardless Bitcoin is very clearly not a mesh network so that part should be changed.

**Comment by UnstoppableCash** [comment]
Additional support for "small-world" network topology [here](https://www.researchgate.net/publication/262562539_Exploring_the_Bitcoin_Network?_esc=publicationCoverPdf&el=1_x_2&enrichId=rgreq-a38ecd7f02c9b45e7a5a47135ba2b9bf-XXX&enrichSource=Y292ZXJQYWdlOzI2MjU2MjUzOTtBUzoxMDAzMTE5ODExMDEwNTZAMTQwMDkyNzY0ODQ5NA%3D%3D)

"... indicating a small world network."

"which is indication for Bitcoin being a small world network as described by Watts and Strogatz"

**Comment by gmaxwell** [comment]
Connecting to everyone else almost always slows down propagation, rather than speeding it. This is because sending data to everyone yourself bottlenecks the transmission on the network links near you (and your high latencies for retransmission to far away hosts). Instead, if the data is sent to a moderate number of nodes, those nodes each send to a moderate number and the amount of bandwidth available to propagate grows exponentially. More transmission also then ends up being across short RTT paths where retransmissions aren't so slow.

While some incompetent miners have at times massively over-connected-- one only need look at the actual network (and the software implementing it) to see that this isn't so generally.  It certainly isn't a property by design or requirement.  No released version of the Bitcoin software ever released has behaved in the way that was being claimed here. (moreover, both "scale free" and "small world" don't mean what you seem to think they mean)

**Comment by UnstoppableCash** [comment]
@gmaxwell 

"Connecting to everyone else..."

Where has my PR said "connecting to everyone else"?

What do you consider "moderate number" of nodes (specifically Miner P2P connections)?

Can you please share data you have showing that "nodes" only have a moderate number of connections?
And to be clear, I am referring to miner node P2P connections in this context.

Thanks!

**Comment by UnstoppableCash** [comment]
@gmaxwell 

Can you show me where I have said "scale free" in the PR or any of my comments?

**Comment by mpapec** [comment]
This is followed by repeatable benchmarks in controled environment, or it didn't happen.

> Connecting to everyone else almost always slows down propagation, rather than speeding it.

**Comment by pzenger** [comment]
Closing this PR for now as it is currently inactive and the project structure has since changed.  
If we want to capture this work, please create an issue.
